app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -0,0 +1,306 @@
+                    newDevices.add(device);
+                    newDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
+                }
+                // When discovery is finished, change the Activity title
+            } else if (BluetoothAdapter.ACTION_DISCOVERY_FINISHED.equals(action)) {
+                setTitle("select_device");
+                if (newDevicesArrayAdapter.getCount() == 0) {
+                    newDevicesArrayAdapter.add("none_found");
+                }
+            }
+
+        }
+    };
+    private ArrayAdapter<String> pairedDevicesArrayAdapter;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Register for broadcasts when a device is discovered.
+        IntentFilter filter = new IntentFilter(BluetoothDevice.ACTION_FOUND);
+        registerReceiver(broadcastReceiver, filter);
+
+        setContentView(R.layout.activity_bluetoothconection);
+
+        // Set result CANCELED in case the user backs out
+        setResult(Activity.RESULT_CANCELED);
+
+        // SetUp bluetooth adapter
+        setUp();
+
+        // Initialize array adapters. One for already paired devices and
+        // one for newly discovered devices
+        pairedDevicesArrayAdapter =
+                new ArrayAdapter<>(this, android.R.layout.simple_list_item_1);
+        newDevicesArrayAdapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1);
+
+        // Find and set up the ListView for paired devices
+        final ListView pairedListView = (ListView) findViewById(R.id.pairedlist);
+        pairedListView.setAdapter(pairedDevicesArrayAdapter);
+        pairedListView.setOnItemClickListener(mDeviceClickListener);
+
+
+        // Find and set up the ListView for newly discovered devices
+        final ListView newDevicesListView = (ListView) findViewById(R.id.discoveredlist);
+        newDevicesListView.setAdapter(newDevicesArrayAdapter);
+        newDevicesListView.setOnItemClickListener(mDeviceClickListener);
+
+
+        // Register for broadcasts when discovery has finished
+        filter = new IntentFilter(BluetoothAdapter.ACTION_DISCOVERY_FINISHED);
+        this.registerReceiver(broadcastReceiver, filter);
+
+        // Get the serviceLocator.
+        final Intent intent = getIntent();
+        final IServiceLocator serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        // Get the BluetoothConnectionService.
+        this.bluetoothConnectionService = (BluetoothConnectionService) serviceLocator.getBluetoothHandler();
+
+        this.newDevices = new HashSet<>();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        // Don't forget to unregister the ACTION_FOUND receiver.
+        breakDown();
```

Comment:
Isn't the `onDestroy` method intended for breakdown? Could you move the code from `breakDown()` to this method?